server/lib/chat/chat-models.ts

```diff
@@ -381,18 +381,131 @@ export async function updateUserPermissions(
   }
 }
 
+export async function countBannedIdentifiersForChannel(
+  {
+    channelId,
+    targetId,
+    filterBrowserprint = true,
+  }: {
+    channelId: SbChannelId
+    targetId: SbUserId
+    filterBrowserprint?: boolean
+  },
+  withClient?: DbClient,
+): Promise<number> {
+  const { client, done } = await db(withClient)
+
+  try {
+    const query = sql`
+      SELECT COUNT(DISTINCT identifier_type) as "matches"
+      FROM channel_identifier_bans cib
+      WHERE cib.channel_id = ${channelId}
+      AND (cib.identifier_type, cib.identifier_hash) IN (
+        SELECT identifier_type, identifier_hash
+        FROM user_identifiers ui
+        WHERE ui.user_id = ${targetId}
+      )
+    `
+
+    if (filterBrowserprint) {
+      query.append(sql`
+        AND cib.identifier_type != 0
+      `)
+    }
+
+    const result = await client.query<{ matches: string }>(query)
+    return result.rows.length > 0 ? Number(result.rows[0].matches) : 0
+  } finally {
+    done()
+  }
+}
+
 export async function banUserFromChannel(
-  channelId: SbChannelId,
-  moderatorId: SbUserId,
-  targetId: SbUserId,
-  reason?: string,
+  {
+    channelId,
+    moderatorId,
+    targetId,
+    reason,
+    automated = false,
+    connectedUsers = [],
+  }: {
+    channelId: SbChannelId
+    moderatorId?: SbUserId
+    targetId: SbUserId
+    reason?: string
```

Comment:
If reason and automated are exclusive, the type should specify things that way. For example:
```ts
{ channelId: SbChannelId, /*...*/, reason: string } | { channelId: SbChannelId, /*...*/, automated: true, connectedUsers: SbUserId[] }
```